mapswipe_workers/mapswipe_workers/firebase_to_postgres/archive_project.py

```diff
@@ -15,24 +15,54 @@ def archive_project(project_ids: list) -> None:
     """
     for project_id in project_ids:
         logger.info("Archive project with the id {0}".format(project_id))
-        logger.info(
-            "Delete results, groups and tasks of project with the id {0}".format(
-                project_id
-            )
-        )
+        logger.info("Delete results of project with the id {0}".format(project_id))
 
         fb_db = auth.firebaseDB()
         fb_db.reference("v2/results/{0}".format(project_id)).set({})
-        fb_db.reference("v2/groups/{0}".format(project_id)).set({})
-        fb_db.reference("v2/tasks/{0}".format(project_id)).set({})
 
+        # get group keys for this project to estimate size in firebase
+        groups = fb_db.reference("v2/groups/{0}/".format(project_id)).get(shallow=True)
+
+        if not groups:
+            logger.info("no groups to delete in firebase")
+        else:
+            group_keys = list(groups.keys())
+            chunk_size = 250
+            chunks = int(len(group_keys) / chunk_size) + 1
+
+            # delete groups, tasks in firebase for each chunk using the update function
+            for i in range(0, chunks):
+                logger.info(
+                    "Delete max {1} groups and tasks of project with the id {0}".format(
+                        project_id, chunk_size
+                    )
+                )
+                update_dict = {}
+                for group_id in group_keys[:chunk_size]:
+                    update_dict[group_id] = None
+                fb_db.reference("v2/groups/{0}".format(project_id)).update(update_dict)
+                fb_db.reference("v2/tasks/{0}".format(project_id)).update(update_dict)
+                group_keys = group_keys[chunk_size:]
+
+        logger.info(
+            "Set status=archived in Firebase for project with the id {0}".format(
+                project_id
+            )
+        )
         fb_db = auth.firebaseDB()
         ref = fb_db.reference("v2/projects/{0}/status".format(project_id))
         ref.set("archived")
 
+        logger.info(
+            "Set status=archived in Postgres for project with the id {0}".format(
+                project_id
+            )
+        )
         pg_db = auth.postgresDB()
-        sql_query = (
-            "UPDATE projects SET status = 'archived' "
-            + "WHERE project_id = '{0}'".format(project_id)
+        sql_query = """
+            UPDATE projects SET status = 'archived'
+            WHERE project_id = '{0}';
+        """.format(
+            project_id
```

Comment:
Are you sure that `project_id` is an integer? Is it possible that anything else can be injected?
Also, you should use f-strings here too, it's much easier to read :)

Reply:
we use save string formatting now and pass named parameters to the query instead of using string formatting